pkg/sql/sessioninit/cache.go

```diff
@@ -291,6 +291,7 @@ func (a *Cache) GetDefaultSettings(
 			// If dbDesc is nil, the database name was not valid, but that should
 			// not cause a login-preventing error.
 			if dbDesc != nil {
+				log.Sessions.Infof(ctx, "provided database, %s, does not exist", databaseName)
```

Comment:
since this is in the `sessions` channel, maybe we could make the message a bit more specific to how it relates to connecting:

```
(... "cannot connect to %s; database does not exist", databaseName)
```

Reply:
sgtm!